iotscape_Beamlogic.py

```diff
@@ -0,0 +1,210 @@
+                        print ("Is SiteAnalyzerAdapter running?")
+                        time.sleep(1)
+        except Exception as err:
+            logCrash(self.name, err)
+
+    #======================== public ==========================================
+
+    #======================== private =========================================
+
+    def _newByte(self, b):
+        """
+        Just received a byte from the sniffer
+        """
+        with self.dataLock:
+            self.rxBuffer += [b]
+            # PCAP global header
+            if   not self.doneReceivingGlobalHeader:
+                if len(self.rxBuffer) == self.PCAP_GLOBALHEADER_LEN:
+                    self.doneReceivingGlobalHeader    = True
+
+                    #Send global header
+                    self.wrThread.publishFrame(self.rxBuffer)
+                    self.rxBuffer                     = []
+
+            # PCAP packet header
+            elif not self.doneReceivingPacketHeader:
+                if len(self.rxBuffer) == self.PCAP_PACKETHEADER_LEN:
+                    self.doneReceivingPacketHeader    = True
+                    self.packetHeader                 = self._parsePcapPacketHeader(self.rxBuffer)
+                    assert self.packetHeader['incl_len'] == self.packetHeader['orig_len']
+
+                    #Append header to packet
+                    self.packet                       += self.rxBuffer
+                    self.rxBuffer                     = []
+                    
+            # PCAP packet bytes
+            else:
+                if len(self.rxBuffer) == self.packetHeader['incl_len']:
+                    self.doneReceivingPacketHeader    = False
+                    self._newFrame(self.rxBuffer)
+                    self.rxBuffer                     = []
+
+
+    def _parsePcapPacketHeader(self, header):
+        """
+        Parse a PCAP packet header
+        Per https://wiki.wireshark.org/Development/LibpcapFileFormat:
+        typedef struct pcaprec_hdr_s {
+            guint32 ts_sec;         /* timestamp seconds */
+            guint32 ts_usec;        /* timestamp microseconds */
+            guint32 incl_len;       /* number of octets of packet saved in file */
+            guint32 orig_len;       /* actual length of packet */
+        } pcaprec_hdr_t;
+        """
+        assert len(header) == self.PCAP_PACKETHEADER_LEN
+
+        returnVal = {}
+        (
+            returnVal['ts_sec'],
+            returnVal['ts_usec'],
+            returnVal['incl_len'],
+            returnVal['orig_len'],
+        ) = struct.unpack('<IIII', ''.join([chr(b) for b in header]))
+
+        return returnVal
+
+
+    def _newFrame(self, frame):
+        """
+        Just received a full frame from the sniffer
+        """
+        self.packet += frame
+        self.wrThread.publishFrame(self.packet)
+
+class WrThread(threading.Thread):
+
+    OUTPUT                   = 'beamlogic'+time.strftime('%Y%m%d-%Hh%Mm%Ss')+'.pcap'
```

Comment:
- rename to `FILENAME_OUTPUT`

Rather than
```
'beamlogic'+time.strftime('%Y%m%d-%Hh%Mm%Ss')+'.pcap'
```
use
```
'beamlogic_{0}.pcap'.format(time.strftime('%Y%m%d-%Hh%Mm%Ss'))
```